src/boot/java/io/github/brassmc/brassloader/boot/discovery/ModDiscovery.java

```diff
@@ -0,0 +1,260 @@
+
+                        // Construct contact
+                        ModContainer.Contact.Builder contact = new ModContainer.Contact.Builder()
+                                .homepage(homepage)
+                                .issues(issues)
+                                .source(source)
+                                .wiki(wiki)
+                                .youtube(youtube)
+                                .twitter(twitter)
+                                .discord(discord);
+
+                        // Construct People
+                        ModContainer.People.Builder people = new ModContainer.People.Builder(developers[0], Arrays.copyOfRange(developers, 1, developers.length))
+                                .artists(artists)
+                                .modellers(modellers)
+                                .animators(animators)
+                                .audioEngineers(audioEngineers)
+                                .additionalCredits(additionalCredits);
+
+                        //Construct container
+                        ModContainer.Builder containerBuilder = new ModContainer.Builder(modid)
+                                .name(name)
+                                .version(version)
+                                .description(description)
+                                .license(license)
+                                .entrypoint(entrypoint)
+                                .people(people)
+                                .icon(icon)
+                                .contact(contact);
+                        if(usesMixins) {
+                            containerBuilder = containerBuilder.usesMixins();
+                        }
+
+                        ModContainer container = containerBuilder.build();
+                        MODS.add(container);
+
+                        // Close resources
+                        stream.close();
+
+                        return FileVisitResult.CONTINUE;
+                    } catch (IOException | ParseException exception) {
+                        exception.printStackTrace();
+                    }
+
+                    return super.visitFile(file, attrs);
+                }
+            });
+
+            return mods.stream().map(mod -> new Resource(IModuleLayerManager.Layer.GAME, List.of(mod))).toList();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    private static String[] getArrayOrString(JsonObject object, String name, Path modLoc, boolean shouldThrowIfMissing) {
+        JsonValue itemsJson = object.get(name);
+        String[] items;
+        if(itemsJson != null && itemsJson.isArray()) {
+            String[] arr = itemsJson.asArray().values().stream().map(JsonValue::asString).toArray(String[]::new);
+            for (int i = 0; i < arr.length; i++) {
+                String author = arr[i];
+                lengthCheck(name + " " + (i + 1), author, modLoc, 4, 32);
```

Comment:
The additional credits use the same method, and they may be structured like  `Thanks to SomeVeryLongUsername for some very long feature.`, which will surpass 32 characters. Please add an int parameter for the limit.